src/compiler/path.ts

```diff
@@ -547,6 +547,10 @@ namespace ts {
 
     export function normalizePath(path: string): string {
         path = normalizeSlashes(path);
+        path = path.replace(/\/\.\//g, "/");

```

Comment:
I could go either way on this replacement.  Among paths that require updates, this is commonly enough (1/3 if I had to ballpark it), but the number is dwarfed by the number that require no changes at all (in which case this is a wasted string traversal).  I measured it both ways and the difference was less than the margin of error.